### ISSUE TITLE
Period in cluster name

### ISSUE THREAD

**Issue by zaglex** [yandex/ClickHouse#275]
I've tried to setup clusters with names 'stat.cluster001' and 'stat.cluster002'.
But distributed tables did not work with such clusternames: select requests were failing. Request "select * from system.clusters" also failed.

I've solved the problem by replacing periods with underscores, but I think that either period should be supported, or such behaviour should be documented.

**Comment by alexey-milovidov** [comment]
Thanks for the report! At least we should check for supported cluster names more early and print meaningful error message.

**Comment by ztlpn** [comment]
It is [the limitation of the Poco configuration reader](https://github.com/ClickHouse-Extras/poco/blob/d7a4383c4d85b51938b62ed5812bc0935245edb3/Util/include/Poco/Util/XMLConfiguration.h#L76). It needs some character to act as the delimiter of hierarchy levels and it is thus forbidden in the configuration keys. The easiest fix is to explicitly forbid dots in cluster names.

The mention in the docs has been added recently: 48d5d03d27592a52a886f99bd3dd5f2e71408f89

Now I've added a clear error message: bd38b4f560e49a72ca108975bd0d976bc23d445c
